### PR TITLE
Move namespace to Doctrine/Collections

### LINKED ISSUE #193: Leave Doctrine/Common namespace

### PR DESCRIPTION
**Pull request by alexander-schranz**
fixes #193

# TODO

 - [ ] Update Tests to new Namespace (currently in old to detect accidently bc breakes)

### CODE REVIEW

File: lib/Doctrine/Collections/ArrayCollection.php [hunk 1]
```diff
@@ -0,0 +1,455 @@
+<?php
+
+namespace Doctrine\Collections;
+
+use ArrayIterator;
+use Closure;
+use Doctrine\Collections\Expr\ClosureExpressionVisitor;
+use Doctrine\Common\Collections\Collection as OldCollection;
```
Comment (alexander-schranz) [inline lib/Doctrine/Collections/ArrayCollection.php:8]: Maybe somebody has a tip here its really strange but without using `OldCollection` which is just a alias to new `Collection` class I get here:

```
TypeError: Return value of Doctrine\Tests\Common\Collections
\ArrayCollectionTest::buildCollection() must be an instance of Doctrine\Common
\Collections\Collection, instance of Doctrine\Collections\ArrayCollection returned
````

and

```
TypeError: Argument 1 passed to Doctrine\Tests\LazyArrayCollection::__construct() 
must be an instance of Doctrine\Common\Collections\Collection, instance of 
Doctrine\Collections\ArrayCollection given
````

I'm not sure why as `Doctrine\Common\Collections\Collection` is an alias to `Doctrine\Collections\Collection` and `ArrayCollection` is extending from it.

**Comment by greg0ire** [comment]
Heads up: I did this on `doctrine/persistence` and it was a very stressful experience. 0/10 would not recommend.
You can read more on this at https://alcaeus.medium.com/how-to-break-an-entire-ecosystem-by-publishing-a-release-b6aaab2b8aaa

**Comment by alcaeus** [comment]
Basically, what @greg0ire said. Been there, done that. Thanks for the contribution, but no thanks.

**Comment by alexander-schranz** [comment]
Then I think we can close https://github.com/doctrine/collections/issues/193

**Comment by alcaeus** [comment]
Done, thanks for pointing it out!

**Comment by alexander-schranz** [comment]
No problem!

**Pull request closed without merging**